nipype/pipeline/plugins/multiproc.py

```diff
@@ -134,53 +111,37 @@ def __init__(self, plugin_args=None):
         self._taskresult = {}
         self._task_obj = {}
         self._taskid = 0
-        non_daemon = True
-        self.plugin_args = plugin_args
-        self.processors = cpu_count()
-        self.memory_gb = get_system_total_memory_gb()*0.9 # 90% of system memory
-
-        self._timeout=2.0
-        self._event = threading.Event()
-
-
+        self._timeout = 2.0
+        # self._event = threading.Event()
 
-        # Check plugin args
-        if self.plugin_args:
-            if 'non_daemon' in self.plugin_args:
-                non_daemon = plugin_args['non_daemon']
-            if 'n_procs' in self.plugin_args:
-                self.processors = self.plugin_args['n_procs']
-            if 'memory_gb' in self.plugin_args:
-                self.memory_gb = self.plugin_args['memory_gb']
-
-        logger.debug("MultiProcPlugin starting %d threads in pool"%(self.processors))
+        # Read in options or set defaults.
+        non_daemon = self.plugin_args.get('non_daemon', True)
+        self.processors = self.plugin_args.get('n_procs', cpu_count())
+        self.memory_gb = self.plugin_args.get('memory_gb',  # Allocate 90% of system memory
+                                              get_system_total_memory_gb() * 0.9)
+        self.raise_insufficient = self.plugin_args.get('raise_insufficient', True)
 
         # Instantiate different thread pools for non-daemon processes
-        if non_daemon:
-            # run the execution using the non-daemon pool subclass
-            self.pool = NonDaemonPool(processes=self.processors)
-        else:
-            self.pool = Pool(processes=self.processors)
-
-    def _wait(self):
-        if len(self.pending_tasks) > 0:
-            if self._config['execution']['poll_sleep_duration']:
-                self._timeout = float(self._config['execution']['poll_sleep_duration'])
-            sig_received=self._event.wait(self._timeout)
-            if not sig_received:
-                logger.debug('MultiProcPlugin timeout before signal received. Deadlock averted??')
-            self._event.clear()
+        logger.debug('MultiProcPlugin starting in "%sdaemon" mode (n_procs=%d, mem_gb=%0.2f)',
+                     'non' if non_daemon else '', self.processors, self.memory_gb)
+        self.pool = (NonDaemonPool(processes=self.processors)
+                     if non_daemon else Pool(processes=self.processors))
```

Comment:
Possibly slightly cleaner, if it can fit on one line:

```Python
self.pool = (NonDaemonPool if non_daemon else Pool)(processes=self.processors)
```

I don't know. Might not be an improvement.